nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -840,6 +862,61 @@ mod test {
         logctx.cleanup_successful();
     }
 
+    #[test]
+    fn test_disk_expungement_removes_zones() {
+        static TEST_NAME: &str = "planner_disk_expungement_removes_zones";
+        let logctx = test_setup_log(TEST_NAME);
+
+        // Create an example system with a single sled
+        let (collection, input, blueprint1) =
+            example(&logctx.log, TEST_NAME, 1);
+
+        let mut builder = input.into_builder();
+
+        // Avoid churning on the quantity of Nexus zones - we're okay staying at
+        // one.
+        builder.policy_mut().target_nexus_zone_count = 1;
+
+        let (_, sled_details) = builder.sleds_mut().iter_mut().next().unwrap();
+        let (_, disk) =
+            sled_details.resources.zpools.iter_mut().next().unwrap();
+        disk.policy =
+            nexus_types::external_api::views::PhysicalDiskPolicy::Expunged;
```

Comment:
Could you just import the `PhysicalDiskPolicy` type?

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/5599/commits/09f416202b93c263aee224fe3a0bee813fff8490